### ISSUE TITLE
Apply compile options for OS X platform when compiling resources

### ISSUE THREAD

**Issue by espakm** [CppMicroServices/CppMicroServices#151]
I see these warnings when I am compiling MITK on El Capitan (11.11)
using the Yosemite system environment (10.10).

```
ld: warning: object file (us_MitkCoreTestDriver/us_resources.o) was
built for newer OSX version (10.11) than being linked (10.10)
```

As @saschazelzer noted, this has been fixed by commit 80c885fd5ce91fea0ba6f99a82673bbe1b1f0790

I am creating this issue only because these two other options would likely be necessary as well:

```
-arch ${CMAKE_OSX_ARCHITECTURES} -mmacosx-version-min
${CMAKE_OSX_DEPLOYMENT_TARGET}
```

**Comment by espakm** [comment]
The '-mmacosx-version-min' option must be used with an '=' sign. So, the final command would look like this:

```
COMMAND ${CMAKE_CXX_COMPILER} -arch ${CMAKE_OSX_ARCHITECTURES} -mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET} -isysroot ${CMAKE_OSX_SYSROOT} -c ${US_CMAKE_RESOURCE_DEPENDENCIES_CPP} -o stub.o
```

**Comment by karthikreddy09** [comment]
@espakm How do you invoke the cmake command ? "-G Unix Makefiles" or "-G Xcode" ? 
Which version of Xcode are you using ?

I have a similar case as yours ... I am building against 10.12 SDK running on 10.11 machine and I do not see this warning in the project's build phase. I can't run anything in my Xcode project without setting the minimum deployment target but the build target was clean.

**Comment by espakm** [comment]
I generate makefiles.

I saw the error because I was using a version that did not have 80c885f. That commit fixes the issue. However, these three options should be set together, normally.

**Comment by saschazelzer** [comment]
I am not an expert on compiler command line options for Darwin. But reading the man pages, I think it makes sense to use these options when generating "linkable" resources, even though we do not generate any code which could be affected by e.g. the `-arch` option. I guess it is just safer to use the same compiler flags that are used for "normal" code when manually creating the `.o` resource object file.

Any comments from OS X exports @CppMicroServices/developers ?

**Comment by karthikreddy09** [comment]
Thanks @espakm !

Yes, I agree these flags should be in the compile line. Cmake is not very good at auto-detecting these flags. I am investigating if these flags are part of CMAKE_CXX_FLAGS on Mac.

**Comment by karthikreddy09** [comment]
Fixed in PR #152

@espakm please try the change 92d62a0c9fbfc4ceb94b29bceeffa7e2145e84e2 and let us know if you see any issues.

**Comment by espakm** [comment]
Just wanted to comment that you will need to add if's to check if the
variables are set, but I saw you added them already. I can confirm that
this works fine. Thank you.
